hypothesis-python/src/hypothesis/searchstrategy/strategies.py

```diff
@@ -284,37 +290,25 @@ def example(self, random=None):
                     "#drawing-interactively-in-tests for more details."
                 )
 
-        from hypothesis import find, settings, Verbosity
-
-        # Conjecture will always try the zero example first. This would result
-        # in us producing the same example each time, which is boring, so we
-        # deliberately skip the first example it feeds us.
-        first = []  # type: list
-
-        def condition(x):
-            if first:
-                return True
-            else:
-                first.append(x)
-                return False
-
-        try:
-            return find(
-                self,
-                condition,
-                random=random,
-                settings=settings(
-                    database=None,
-                    verbosity=Verbosity.quiet,
-                    phases=tuple(set(Phase) - {Phase.shrink}),
-                ),
+        if not self.__examples:
+            from hypothesis.core import given
+
+            @given(self)
+            @settings(
+                database=None,
+                max_examples=10,
+                deadline=None,
+                verbosity=Verbosity.quiet,
+                phases=(Phase.generate,),
+                suppress_health_check=HealthCheck.all(),
             )
-        except (NoSuchExample, Unsatisfiable):
-            # This can happen when a strategy has only one example. e.g.
-            # st.just(x). In that case we wanted the first example after all.
-            if first:
-                return first[0]
-            raise NoExamples(u"Could not find any valid examples in 100 tries")
+            def engine_inside_example(ex):
+                self.__examples.append(ex)
+
+            self.__examples = []
+            engine_inside_example()
+            self.__examples.reverse()
+        return self.__examples.pop()
```

Comment:
I think it might be worth shuffling the examples at the point of generation. This means you get a better representation of what given actually does, and it guards against the case where you only try it once (in which case it always gives you the minimal example) either deliberately or because you're trying a strategy that is recreated each time.

In fact, I'm not sure I'm keen on keeping state around on the strategy objects to support what is a very niche use case whose performance mostly shouldn't matter. `10` isn't a very large number, so why not just generate all `10` each time and return a randomly sampled one?